examples/cmd/testing/recordingscript.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/src-d/metadata-retrieval/github"
+
+	"github.com/motemen/go-loghttp"
+	"golang.org/x/oauth2"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+// This script will use the downloader to crawl using the graphql API a repository (default: src-d/gitbase)
+// and an organization (default: src-d) and will record requests (graphql queries) and responses (json body data)
```

Comment:
are you sure `src-d` is a good choice? it's a big organization. I'm not even sure if we can download it all with only one token without hitting the limit.

Reply:
At least for what the `DownloadOrganization` method crawls for `src-d` the gob file is 31KB, while for the `DownloadRepository` method for `src-d/gitbase` the gob file is 5MB. I wanted a big org/big repo to hit pagination limits, but I will search further to find just the right amout.

Reply:
ah. sorry. `DownloadOrganization` downloads only org info. Not org + everything. I messed-up with ghdeep command.
Sorry, discard my comment.

Reply:
if you want to hit the limit you can do N requests just to reduce number of available calls and then call `DownloadRepository`.
Pseudo-code:
```go
ac := getAvailableCalls(token)
if ac < 20 {
   panic("not enough calls")
}
for ac > 20 {
// do throw-away requests
}
d.DownloadRepository()
```

or just record error response once and return it in your transport when you want instead of real response.

Reply:
Opened #40